api/listener/controller.go

```diff
@@ -151,3 +156,33 @@ func deleteById(c *fiber.Ctx) error {
 
 	return c.JSON(result)
 }
+
+func upsertObservedBlock(c *fiber.Ctx) error {
+	payload := new(ListenerObservedBlockModel)
+	if err := c.BodyParser(payload); err != nil {
+		return err
+	}
+
+	validate := validator.New()
+	validate.RegisterValidation("blockNumberValidator", func(fl validator.FieldLevel) bool {
+		return fl.Field().Int() >= 0
```

Comment:
that's the default validator behavior but I kept running into validation error because locally the first block starts from zero. So, while `fl.Field().Int() > 0` is good in production it wont work locally